cpp/src/strings/repeat_strings.cu

```diff
@@ -166,28 +166,169 @@ std::unique_ptr<column> repeat_strings(strings_column_view const& input,
+    }
+
+    if (d_chars && repeat_times > 0) {
+      auto const d_str    = strings_dv.element<string_view>(idx);
+      auto const str_size = d_str.size_bytes();
+      if (str_size > 0) {
+        auto const input_ptr = d_str.data();
+        auto output_ptr      = d_chars + d_offsets[idx];
+        for (size_type repeat_idx = 0; repeat_idx < repeat_times; ++repeat_idx) {
+          output_ptr = copy_and_increment(output_ptr, input_ptr, str_size);
+        }
+      }
+    }
+  }
+};
+
+}  // namespace
+
+std::unique_ptr<column> repeat_strings(strings_column_view const& strings,
+                                       column_view const& repeat_times,
+                                       rmm::cuda_stream_view stream,
+                                       rmm::mr::device_memory_resource* mr)
+{
+  CUDF_EXPECTS(repeat_times.type().id() == type_id::INT32,
+               "The input repeat_times column must have INT32 data type.");
+  CUDF_EXPECTS(strings.size() == repeat_times.size(), "The input columns must have the same size.");
+
+  auto const strings_count = strings.size();
+  if (strings_count == 0) { return make_empty_column(data_type{type_id::STRING}); }
+
+  auto const strings_dv_ptr      = column_device_view::create(strings.parent(), stream);
+  auto const repeat_times_dv_ptr = column_device_view::create(repeat_times, stream);
+  auto const strings_has_nulls   = strings.has_nulls();
+  auto const rtimes_has_nulls    = repeat_times.has_nulls();
+  auto const fn                  = compute_size_and_repeat_separately_fn{
+    *strings_dv_ptr, *repeat_times_dv_ptr, strings_has_nulls, rtimes_has_nulls};
+
+  // Repeat the strings in each row.
+  // Note that this cannot handle the cases when the size of the output column exceeds the maximum
+  // value that can be indexed by size_type (offset_type).
+  // In such situations, an exception may be thrown, or the output result is undefined.
+  auto [offsets_column, chars_column, null_mask, null_count] = [&] {
+    // If both input columns have nulls, we need to generate a new null mask.
+    if (strings_has_nulls && rtimes_has_nulls) {
+      return make_strings_children_with_null_mask(fn, strings_count, strings_count, stream, mr);
+    }
+
+    // Generate output strings without null mask.
+    auto [offsets_column, chars_column] = make_strings_children(fn, strings_count, stream, mr);
+
+    // If only one input column has null, we just copy its null mask and null count.
+    if (strings_has_nulls ^ rtimes_has_nulls) {
+      auto const& col = strings_has_nulls ? strings.parent() : repeat_times;
+      return std::make_tuple(std::move(offsets_column),
+                             std::move(chars_column),
+                             cudf::detail::copy_bitmask(col, stream, mr),
+                             col.null_count());
+    }
+
+    // Both input columns do not have nulls.
+    return std::make_tuple(
+      std::move(offsets_column), std::move(chars_column), rmm::device_buffer{}, 0);
```

Comment:
We still want to use the right stream and memory resource here.
```suggestion
      std::move(offsets_column), std::move(chars_column), rmm::device_buffer{0, stream, mr}, 0);
```

Reply:
Can you explain more, please? Why a zero size buffer stills need them?

Reply:
`device_buffer` keeps a copy of the stream and memory resource. I'm not sure that it affects the behavior when creating and using a zero-size buffer, but it might.